sys/posix/pthread/pthread.c

```diff
@@ -160,7 +161,7 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *attr, void *(*sta
                                    pthread_start_routine,
                                    pt,
                                    "pthread");
-    if (pt->thread_pid == KERNEL_PID_UNDEF) {
+    if (pt->thread_pid == -EOVERFLOW) {
```

Comment:
in error case `thread_create` can return `-EAGAIN`, `-EOVERFLOW` or `KERNEL_PID_UNDEF`, hence I would suggest to check for `<= KERNEL_PID_UNDEF`?

Reply:
I dont think it can return `KERNEL_PID_UNDEF` because it has 3 returns
Code from core/thread.c
```
if (pid == KERNEL_PID_UNDEF) {
        DEBUG("thread_create(): too many threads!\n");

        irq_restore(state);

        return -EOVERFLOW;
    }
```

`return pid;` cannot return `KERNEL_PID_UNDEF` because of previous code `if (pid == KERNEL_PID_UNDEF) { ... return -EOVERFLOW`

and this return that only returns when the priority is wrong. But the priority is static (always  `THREAD_PRIORITY_MAIN`) so this should never return
```
if (priority >= SCHED_PRIO_LEVELS) {
        return -EINVAL;
    }
```

That is why I only check for `-EOVERFLOW`
But the check `<= KERNEL_PID_UNDEF` could be easier to read. 
Would you still suggest the change?

Reply:
mhm, maybe check for ` < KERNEL_PID_UNDEF` then, bc I wouldn't rely on the priority to work either - e.g. someone might redefine `THREAD_PRIORITY_MAIN` making it invalid.

However, you're right it can never return `KERNEL_PID_UNDEF`.

Reply:
I will use `pid_is_valid` that should be the right way to do it.